tests/expressions_window/tests.py

```diff
@@ -0,0 +1,801 @@
+            ('Johnson', 'Management', 80000, 1),
+            ('Wilkinson', 'IT', 60000, 1),
+            ('Smith', 'Sales', 55000, 2),
+            ('Brown', 'Sales', 53000, 2),
+            ('Adams', 'Accounting', 50000, 2),
+            ('Jenson', 'Accounting', 45000, 3),
+            ('Jones', 'Accounting', 45000, 3),
+            ('Johnson', 'Marketing', 40000, 3),
+            ('Smith', 'Marketing', 38000, 4),
+            ('Williams', 'Accounting', 37000, 4),
+            ('Moore', 'IT', 34000, 4),
+        ], lambda x: (x.name, x.department, x.salary, x.ntile))
+
+    def test_percent_rank(self):
+        """
+        Calculate the percentage rank of the employees across the entire
+        company based on salary and name (in case of ambiguity).
+        """
+        qs = Employee.objects.annotate(percent_rank=Window(
+            expression=PercentRank(),
+            order_by=[F('salary').asc(), F('name').asc()],
+        )).order_by('percent_rank')
+
+        # round for test unification, Oracle and PostgreSQL
+        self.assertQuerysetEqual(qs, [
+            ('Moore', 'IT', 34000, 0.0),
+            ('Williams', 'Accounting', 37000, 0.0909090909),
+            ('Smith', 'Marketing', 38000, 0.1818181818),
+            ('Johnson', 'Marketing', 40000, 0.2727272727),
+            ('Jenson', 'Accounting', 45000, 0.3636363636),
+            ('Jones', 'Accounting', 45000, 0.4545454545),
+            ('Adams', 'Accounting', 50000, 0.5454545455),
+            ('Brown', 'Sales', 53000, 0.6363636364),
+            ('Smith', 'Sales', 55000, 0.7272727273),
+            ('Wilkinson', 'IT', 60000, 0.8181818182),
+            ('Johnson', 'Management', 80000, 0.9090909091),
+            ('Miller', 'Management', 100000, 1.0),
+        ], transform=lambda row: (row.name, row.department, row.salary, round(row.percent_rank, 10)))
+
+    def test_nth_returns_null(self):
+        """
+        Find the nth row of the data set. There are fewer than 20 rows in the
+        test data, which should return null.
+        """
+        qs = Employee.objects.annotate(nth_value=Window(
+            expression=NthValue('salary', nth=20),
+            order_by=F('salary').asc()
+        )).order_by('salary', 'name')
+
+        self.assertQuerysetEqual(qs, [
+            ('Moore', 34000, 'IT', None),
+            ('Williams', 37000, 'Accounting', None),
+            ('Smith', 38000, 'Marketing', None),
+            ('Johnson', 40000, 'Marketing', None),
+            ('Jenson', 45000, 'Accounting', None),
+            ('Jones', 45000, 'Accounting', None),
+            ('Adams', 50000, 'Accounting', None),
+            ('Brown', 53000, 'Sales', None),
+            ('Smith', 55000, 'Sales', None),
+            ('Wilkinson', 60000, 'IT', None),
+            ('Johnson', 80000, 'Management', None),
+            ('Miller', 100000, 'Management', None),
+        ], lambda row: (row.name, row.salary, row.department, row.nth_value))
```

Comment:
It seems that ordering is an issue, therefore:
```python
qs = Employee.objects.annotate(nth_value=Window(
    expression=NthValue('salary', nth=20),
    order_by=F('salary').asc()
))
self.assertListEqual(list(qs.values_list('nth_value', flat=True).distinct()), [None])
```
will work properly.

Reply:
Thanks :)